util/EIP-1024/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+extern crate saltbabe;
+#[macro_use]
+extern crate arrayref;
+extern crate hex;
+use std::str;
+pub use saltbabe::{KeyPair,Public, Secret, Error};
+pub use saltbabe::traits::FromUnsafeSlice;
+
+const VERSION: &str =  "x25519-xsalsa20-poly1305";
+
+#[derive(Debug, Clone)]
+pub struct EncryptedData {
+    version: String,    
+    nonce: String,
+    ephem_public: String,
+    ciphertext: String
+}
+
+pub fn get_encryption_keypair(sk: [u8; 32]) -> KeyPair<Secret, Public> {
+    let keypair = saltbabe::crypto_box::gen_keypair_from_secret(&sk);
+    return keypair;
+}
+
+pub fn to_byte32(bytes: &[u8]) -> [u8; 32] {
+    array_ref!(bytes, 0, 32).clone()
+}
+
+
+pub fn encrypt(data: &[u8], version: Option<String>, send_pk: [u8; 32], ephermal_sk: [u8; 32]) -> Result<EncryptedData, Error> {
```

Comment:
Document this function because it is not obvious to users of this API that `x25519-xsalsa20-poly1305" will be used if version is `None`.

Also, the line is too long

Reply:
ok